scripts/prepare.js

```diff
@@ -14,28 +14,15 @@ if (process.env.npm_config_zmq_external == "true") {
 
 function buildZMQ(scriptPath, zmqDir) {
   console.log("Building libzmq for " + process.platform);
-  try {
-    const child = spawnSync(scriptPath, [ZMQ, ARCH], {
-      cwd: process.cwd(),
-      env: process.env,
-      stdio: ['inherit', 'inherit', 'pipe'],
-      encoding: 'utf-8',
-    });
-    if (child.stderr) {
-      return console.log(Error(child.stderr))
-    }
-    if (child.status !== 0) {
-      return console.error("child process exited with code " + child.status);
-    }
-  } catch(e) {
-    return console.error(e)
+  const child = spawnSync("/bin/sh", [scriptPath, ZMQ, ARCH], {
```

Comment:
I hope this doesn't break when `scriptPath` has spaces in its names.

Reply:
> I hope this doesn't break when scriptPath has spaces in its names.

This is one of the main differences between `execSync` (we pass a command) and `spawnSync` (we pass arguments).

I went ahead and test if `spawnSync` is working as expected. See below: It does, but other things fail:
 * [`node-gyp` doesn't work with path containing spaces](https://github.com/nodejs/node-gyp/blob/4b83c3de7300457919d53f26d96ea9ad6f6bedd8/gyp/pylib/gyp/generator/make.py#L984-L989).
 * we should quote the following assignments in `build_libzmq.sh`: https://github.com/n-riesco/zeromq.js/blob/f7a437b7453cea63c874d8771fd5cf0e12cbaf02/scripts/build_libzmq.sh#L25-L26

```
$ pwd
/home/nriesco/jupyter/github/zeromq js
$ rm -rf build/Release zmq
$ npm i

> zeromq@5.2.7 install /home/nriesco/jupyter/github/zeromq js
> node-gyp-build || npm run build:libzmq

make: Entering directory '/home/nriesco/jupyter/github/zeromq js/build'
  ACTION binding_gyp_libzmq_target_prepare_build zmq/BUILD_SUCCESS
/bin/sh: 1: js/build/Release/lib.host:/home/nriesco/jupyter/github/zeromq: not found
Building libzmq for linux
build_libzmq.sh: Building version 4.2.2 for architecture x64
/home/nriesco/jupyter/github/zeromq js/scripts/build_libzmq.sh: 22: export: js/scripts: bad variable name

[...]
```